ecosystem/typescript/sdk/src/plugins/ans_client.ts

```diff
@@ -94,6 +96,39 @@ export class AnsClient {
     return this.getAddressByDomainName(domain);
   }
 
+  /**
+   * Mint a new Aptos name
+   *
+   * @param account AptosAccount where collection will be created
+   * @param domainName Aptos domain name to mint
+   * @param years year duration of the domain name
+   * @returns The hash of the pending transaction submitted to the API
+   */
+  async mintAptosName(account: AptosAccount, domainName: string, years: number = 1): Promise<Gen.HashValue> {
+    // check if the name is valid
+    if (domainName.match(nameComponentPattern) === null) {
+      throw new ApiError(
+          400,
+          `Name ${domainName} is not valid`,
+      );
+    }
+    // check if the name is available
+    const address = await this.getAddressByName(domainName);
+    if (address !== null) {
+      throw new ApiError(
```

Comment:
do we need it here? `this.getAddressByName` uses `getTableItem` that has `@parseApiError` decorator - I would assumed that this decorator handles api errors

Reply:
we throw this error when `this.getAddressByName(domainName)` return an address. it's not an error for `this.getAddressByName(domainName)` but we are not suppose to mint name in that case. Thus we throw error here.

Reply:
got it. thanks!